src/clj/puppetlabs/puppetserver/certificate_authority.clj

```diff
@@ -1168,6 +1168,31 @@
          (whitelist-matches? autosign subject))
        false))))
 
+(schema/defn ensure-cn-as-san :- utils/SSLExtension
+  "Given an SSLExtension for alt names and a cn, ensure that the cn is listed as a DNS alt name."
+  [extension :- utils/SSLExtension
+   cn :- schema/Str]
+  (if (some #(= cn %) (get-in extension [:value :dns-name]))
+    extension
+    (update-in extension [:value :dns-name] conj cn)))
+
+(schema/defn is-san?
+  [extension]
+  (= utils/subject-alt-name-oid (:oid extension)))
+
+(schema/defn ensure-ext-list-has-cn-san
+  "Given a list of extensions to be signed onto a certificate, ensure that a CN is provided
+   as a subject alternative name; if no subject alternative name is found, generate a new
+   extension and add it to the list with the CN supplied"
+  [cn :- schema/Str
+   extensions :- (schema/pred utils/extension-list?)]
+  (let [[san] (filter is-san? extensions)
```

Comment:
Why is this wrapped in an extra set of brackets?

Reply:
This is the equivalent of this: ```(let [san (first (filter is-san? extensions))]```

The ```[san]``` just tells it to return the first element of that sequence returned by the filter.

Reply:
oooh interesting! Thanks for explaining!